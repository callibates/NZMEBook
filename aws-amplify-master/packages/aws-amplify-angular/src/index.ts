export { AmplifyAngularModule } from './aws-amplify-angular.module';

export * from './components';
export * from './providers';
