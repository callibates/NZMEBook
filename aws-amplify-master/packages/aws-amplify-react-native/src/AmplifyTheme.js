/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import { StyleSheet } from 'react-native';
export default AmplifyTheme = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'space-around',
        paddingTop: 20,
        width: '100%'
    },
    section: {
        flex: 1,
        width: '100%'
    },
    sectionHeader: {
    },
    sectionHeaderText: {
        width: '100%',
        padding: 10,
        textAlign: 'center',
        backgroundColor: '#007bff',
        color: '#ffffff',
        fontSize: 20,
        fontWeight: '500'
    },
    sectionFooter: {
        width: '100%',
        marginTop: 15,
        padding: 10,
        flexDirection: 'row',
        justifyContent: 'space-between'
    },
    sectionFooterLink: {
        fontSize: 14,
        color: '#007bff',
        alignItems: 'baseline',
        textAlign: 'center'
    },
    sectionBody: {
    },
    cell: {
        flex: 1,
        width: '50%'
    },
    errorRow: {
        flex: 1
    },

    photo: {
        width: '100%'
    },
    album: {
        width: '100%'
    },

    a: {
    },
    button: {
        backgroundColor: '#007bff'
    },

    input: {
        margin: 6
    }
});

