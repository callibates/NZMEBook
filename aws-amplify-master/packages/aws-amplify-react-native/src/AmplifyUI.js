/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import React from 'react';
import { View, Text, TextInput, TouchableHighlight } from 'react-native';

import { I18n } from 'aws-amplify';

export const Username = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <TextInput
            style={theme.input}
            placeholder={I18n.get('Username')}
            autoFocus={true}
            autoCapitalize="none"
            {...props}
        />
    )
}

export const Password = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <TextInput
            style={theme.input}
            placeholder={I18n.get('Password')}
            secureTextEntry={true}
            {...props}
        />
    )
}

export const Email = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <TextInput
            style={theme.input}
            placeholder={I18n.get('Email')}
            keyboardType="email-address"
            autoCapitalize="none"
            {...props}
        />
    )
}

export const PhoneNumber = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <TextInput
            style={theme.input}
            placeholder={I18n.get('Phone Number')}
            keyboardType="phone-pad"
            {...props}
        />
    )
}

export const ConfirmationCode = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <TextInput
            style={theme.input}
            placeholder={I18n.get('Code')}
            autoFocus={true}
            {...props}
        />
    )
}

export const LinkCell = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <View style={theme.cell}>
            <TouchableHighlight
                onPress={props.onPress}
            >
                <Text style={theme.sectionFooterLink}>{props.children}</Text>
            </TouchableHighlight>
        </View>
    )
}

export const Header = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <View style={theme.sectionHeader}>
            <Text style={theme.sectionHeaderText}>{props.children}</Text>
        </View>
    )
}

export const ErrorRow = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <View style={theme.errorRow}>
            <Text style={theme.erroRowText}>{props.children}</Text>
        </View>
    )
}
