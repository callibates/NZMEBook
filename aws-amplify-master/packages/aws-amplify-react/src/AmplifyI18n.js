/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

const dict = {
    'fr': {
        'Sign In': "Se connecter",
        'Sign Up': "S'inscrire",
        'Sign Out': "Déconnexion",
        'Forgot Password': "Mot de passe oublié",
        'Username': "Nom d'utilisateur",
        'Password': "Mot de passe",
        'Change Password': "Changer le mot de passe",
        'New Password': "nouveau mot de passe",
        'Email': "Email",
        'Phone Number': "Numéro de téléphone",
        'Confirm a Code': "Confirmer un code",
        'Confirm Sign In': "Confirmer la connexion",
        'Confirm Sign Up': "Confirmer l'inscription",
        'Back to Sign In': "Retour à la connexion",
        'Send Code': "Envoyer le code",
        'Confirm': "Confirmer",
        'Resend a Code': "Renvoyer un code",
        'Submit': "Soumettre",
        'Skip': "Sauter",
        'Verify': "Vérifier",
        'Verify Contact': "Vérifier le contact",
        'Code': "Code",
        'Account recovery requires verified contact information':
"La récupération du compte nécessite des informations de contact vérifiées",

        'User does not exist': "L'utilisateur n'existe pas",
        'User already exists': "L'utilisateur existe déjà",
        'Incorrect username or password': "identifiant ou mot de passe incorrect",
        'Invalid password format': "format de mot de passe invalide",
        'Invalid phone number format':
`Format de numéro de téléphone invalide.
Veuillez utiliser un format de numéro de téléphone du +12345678900`
    },

    'es': {
        'Sign In': "Registrarse",
        'Sign Up': "Regístrate",
        'Sign Out': "Desconectar",
        'Forgot Password': "Se te olvidó tu contraseña",
        'Username': "Nombre de usuario",
        'Password': "Contraseña",
        'Change Password': "Cambia la contraseña",
        'New Password': "Nueva contraseña",
        'Email': "Email",
        'Phone Number': "Número de teléfono",
        'Confirm a Code': "Confirmar un código",
        'Confirm Sign In': "Confirmar inicio de sesión",
        'Confirm Sign Up': "Confirmar Registrarse",
        'Back to Sign In': "Volver a Iniciar sesión",
        'Send Code': "Enviar código",
        'Confirm': "Confirmar",
        'Resend a Code': "Reenviar un código",
        'Submit': "Enviar",
        'Skip': "Omitir",
        'Verify': "Verificar",
        'Verify Contact': "Verificar contacto",
        'Code': "Código",
        'Account recovery requires verified contact information':
"La recuperación de la cuenta requiere información de contacto verificada",

        'User does not exist': "el usuario no existe",
        'User already exists': "El usuario ya existe",
        'Incorrect username or password': "Nombre de usuario o contraseña incorrecta",
        'Invalid password format': "Formato de contraseña inválido",
        'Invalid phone number format':
`Formato de número de teléfono inválido.
Utilice el formato de número de teléfono +12345678900`
    },

    'zh': {
        'Sign In': "登录",
        'Sign Up': "注册",
        'Sign Out': "退出",
        'Forgot Password': "忘记密码",
        'Username': "用户名",
        'Password': "密码",
        'Change Password': "改变密码",
        'New Password': "新密码",
        'Email': "邮箱",
        'Phone Number': "电话",
        'Confirm a Code': "确认码",
        'Confirm Sign In': "确认登录",
        'Confirm Sign Up': "确认注册",
        'Back to Sign In': "回到登录",
        'Send Code': "发送确认码",
        'Confirm': "确认",
        'Resend a Code': "重发确认码",
        'Submit': "提交",
        'Skip': "跳过",
        'Verify': "验证",
        'Verify Contact': "验证联系方式",
        'Code': "确认码",
        'Account recovery requires verified contact information':
"账户恢复需要验证过的联系方式",

        'User does not exist': "用户不存在",
        'User already exists': "用户已经存在",
        'Incorrect username or password': "用户名或密码错误",
        'Invalid password format': "密码格式错误",
        'Invalid phone number format': "电话格式错误，请使用格式 +12345678900"
    }
}

export default dict;
