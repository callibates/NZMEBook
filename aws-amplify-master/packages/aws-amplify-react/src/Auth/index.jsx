/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Authenticator from './Authenticator';

export { default as Authenticator } from './Authenticator';
export { default as AuthPiece } from './AuthPiece';
export { default as SignIn } from './SignIn';
export { default as ConfirmSignIn } from './ConfirmSignIn';
export { default as SignOut } from './SignOut';
export { default as RequireNewPassword } from './RequireNewPassword';
export { default as SignUp } from './SignUp';
export { default as ConfirmSignUp } from './ConfirmSignUp';
export { default as VerifyContact } from './VerifyContact';
export { default as ForgotPassword } from './ForgotPassword';
export { default as Greetings } from './Greetings';
export { default as FederatedSignIn, FederatedButtons } from './FederatedSignIn';
export { default as TOTPSetup } from './TOTPSetup';

export * from './Provider';

import Greetings from './Greetings';

export function withAuthenticator(Comp, includeGreetings=false, authenticatorComponents = []) {
    return class extends Component {
        constructor(props) {
            super(props);

            this.handleAuthStateChange = this.handleAuthStateChange.bind(this);

            this.state = {
                authState: props.authState || null,
                authData: props.authData || null
            };
        }

        handleAuthStateChange(state, data) {
            this.setState({ authState: state, authData: data });
        }

        render() {
            const { authState, authData } = this.state;
            const signedIn = (authState === 'signedIn');
            if (signedIn) {
                return (
                    <div>
                        {
                            includeGreetings?
                            <Greetings
                                authState={authState}
                                authData={authData}
                                onStateChange={this.handleAuthStateChange}
                            />
                            : null
                        }
                        <Comp
                            {...this.props}
                            authState={authState}
                            authData={authData}
                            onStateChange={this.handleAuthStateChange}
                        />
                    </div>
                )
            }

            return <Authenticator
                {...this.props}
                hideDefault={authenticatorComponents.length > 0}
                onStateChange={this.handleAuthStateChange}
                children={authenticatorComponents}
            />
        }
    }
}

export class AuthenticatorWrapper extends Component {
    constructor(props) {
        super(props);

        this.state = { auth: 'init' };

        this.handleAuthState = this.handleAuthState.bind(this);
    }

    handleAuthState(state, data) {
        this.setState({ auth: state, authData: data });
    }

    render() {
        return (
            <div>
                <Authenticator {...this.props} onStateChange={this.handleAuthState} />
                {this.props.children(this.state.auth)}
            </div>
        )
    }
}

AuthenticatorWrapper.propTypes = {
    children: PropTypes.func.isRequired
}
