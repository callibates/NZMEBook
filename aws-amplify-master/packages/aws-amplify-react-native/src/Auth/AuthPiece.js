/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import React from 'react';

import { 
    Auth,
    Logger 
} from 'aws-amplify';

import AmplifyTheme from '../AmplifyTheme';
import AmplifyMessageMap from '../AmplifyMessageMap';

const logger = new Logger('AuthPiece');

export default class AuthPiece extends React.Component {
    constructor(props) {
        super(props);

        this._isHidden = true;
        this._validAuthStates = [];
        this.changeState = this.changeState.bind(this);
        this.error = this.error.bind(this);
    }

    changeState(state, data) {
        if (this.props.onStateChange) {
            this.props.onStateChange(state, data);
        }
    }

    error(err) {
        logger.debug(err);

        let msg = '';
        if (typeof err === 'string') {
            msg = err;
        } else if (err.message) {
            msg = err.message;
        } else {
            msg = JSON.stringify(err);
        }

        const map = this.props.errorMessage || AmplifyMessageMap;
        msg = (typeof map === 'string')? map : map(msg);
        this.setState({ error: msg });
    }

    render() {
        if (!this._validAuthStates.includes(this.props.authState)) {
            this._isHidden = true;
            return null;
        }

        if (this._isHidden) {
            const { track } = this.props;
            if (track) track();
        }
        this._isHidden = false;

        return this.showComponent(this.props.theme || AmplifyTheme);
    }

    showComponent(theme) {
        throw 'You must implement showComponent(theme) and don\'t forget to set this._validAuthStates.';
    }
}
