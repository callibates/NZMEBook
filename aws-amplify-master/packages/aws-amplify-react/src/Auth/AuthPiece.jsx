/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import React, { Component } from 'react';
import { Logger } from 'aws-amplify';

import AmplifyTheme from '../AmplifyTheme';
import AmplifyMessageMap from '../AmplifyMessageMap';

const logger = new Logger('AuthPiece');

export default class AuthPiece extends Component {
    constructor(props) {
        super(props);

        this.inputs = {};

        this._isHidden = true;
        this._validAuthStates = [];
        this.changeState = this.changeState.bind(this);
        this.error = this.error.bind(this);
        this.handleInputChange = this.handleInputChange.bind(this);
    }

    // extract username from authData
    usernameFromAuthData() {
        const { authData } = this.props;
        if (!authData) { return ''; }

        let username = '';
        if (typeof authData === 'object') { // user object
            username = authData.user? authData.user.username : authData.username;
        } else {
            username = authData; // username string
        }

        return username;
    }

    errorMessage(err) {
        if (typeof err === 'string') { return err; }
        return err.message? err.message : JSON.stringify(err);
    }

    triggerAuthEvent(event) {
        const state = this.props.authState;
        if (this.props.onAuthEvent) { this.props.onAuthEvent(state, event); }
    }

    changeState(state, data) {
        if (this.props.onStateChange) { this.props.onStateChange(state, data); }

        this.triggerAuthEvent({
            type: 'stateChange',
            data: state
        });
    }

    error(err) {
        this.triggerAuthEvent({
            type: 'error',
            data: this.errorMessage(err)
        });
    }

    handleInputChange(evt) {
        this.inputs = this.inputs || {};
        const { name, value, type, checked } = evt.target;
        const check_type = ['radio', 'checkbox'].includes(type);
        this.inputs[name] = check_type? checked : value;
        this.inputs['checkedValue'] = check_type? value: null;
    }

    render() {
        if (!this._validAuthStates.includes(this.props.authState)) {
            this._isHidden = true;
            return null;
        }

        if (this._isHidden) {
            const { track } = this.props;
            if (track) track();
        }
        this._isHidden = false;

        return this.showComponent(this.props.theme || AmplifyTheme);
    }

    showComponent(theme) {
        throw 'You must implement showComponent(theme) and don\'t forget to set this._validAuthStates.';
    }
}
