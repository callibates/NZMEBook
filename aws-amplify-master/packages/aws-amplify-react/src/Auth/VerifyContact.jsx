/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import React, { Component } from 'react';
import { Auth, I18n, Logger } from 'aws-amplify';

import AuthPiece from './AuthPiece';
import AmplifyTheme from '../AmplifyTheme';
import {
    FormSection,
    SectionHeader,
    SectionBody,
    SectionFooter,
    InputRow,
    RadioRow,
    MessageRow,
    ButtonRow,
    Link
} from '../AmplifyUI';

const logger = new Logger('VerifyContact');

export default class VerifyContact extends AuthPiece {
    constructor(props) {
        super(props);

        this._validAuthStates = ['verifyContact'];
        this.verify = this.verify.bind(this);
        this.submit = this.submit.bind(this);

        this.state = { verifyAttr: null }
    }

    verify() {
        const { contact, checkedValue } = this.inputs;
        if (!contact) {
            this.error('Neither Email nor Phone Number selected');
            return;
        }

        Auth.verifyCurrentUserAttribute(checkedValue)
            .then(data => {
                logger.debug(data);
                this.setState({ verifyAttr: checkedValue });
            })
            .catch(err => this.error(err));
    }

    submit() {
        const attr = this.state.verifyAttr;
        const { code } = this.inputs;
        Auth.verifyCurrentUserAttributeSubmit(attr, code)
            .then(data => {
                logger.debug(data);
                this.changeState('signedIn', this.props.authData);
                this.setState({ verifyAttr: null });
            })
            .catch(err => this.error(err));
    }

    verifyView() {
        const user = this.props.authData;
        if (!user) {
            logger.debug('no user for verify');
            return null;
        }
        const { unverified } = user;
        if (!unverified) {
            logger.debug('no unverified on user');
            return null;
        }
        const { email, phone_number } = unverified;
        const theme = this.props.theme || AmplifyTheme;
        return (
            <div>
                { email? <RadioRow
                            placeholder={I18n.get('Email')}
                            theme={theme}
                            key="email"
                            name="contact"
                            value="email"
                            onChange={this.handleInputChange}
                         /> : null
                }
                { phone_number? <RadioRow
                                    placeholder={I18n.get('Phone Number')}
                                    theme={theme}
                                    key="phone_number"
                                    name="contact"
                                    value="phone_number"
                                    onChange={this.handleInputChange}
                                /> : null
                }
                <ButtonRow theme={theme} onClick={this.verify}>{I18n.get('Verify')}</ButtonRow>
            </div>
        )
    }

    submitView() {
        const theme = this.props.theme || AmplifyTheme;
        return (
            <div>
                <InputRow
                    placeholder={I18n.get('Code')}
                    theme={theme}
                    key="code"
                    name="code"
                    autoComplete="off"
                    onChange={this.handleInputChange}
                />
                <ButtonRow theme={theme} onClick={this.submit}>{I18n.get('Submit')}</ButtonRow>
            </div>
        )
    }

    showComponent(theme) {
        const { authData, hide } = this.props;
        if (hide && hide.includes(VerifyContact)) { return null; }

        return (
            <FormSection theme={theme}>
                <SectionHeader theme={theme}>{I18n.get('Verify Contact')}</SectionHeader>
                <SectionBody theme={theme}>
                    <MessageRow theme={theme}>
                        {I18n.get('Account recovery requires verified contact information')}
                    </MessageRow>
                    { this.state.verifyAttr? this.submitView() : this.verifyView() }
                </SectionBody>
                <SectionFooter theme={theme}>
                    <Link theme={theme} onClick={() => this.changeState('signedIn', authData)}>
                        {I18n.get('Skip')}
                    </Link>
                </SectionFooter>
            </FormSection>
        )
    }
}
