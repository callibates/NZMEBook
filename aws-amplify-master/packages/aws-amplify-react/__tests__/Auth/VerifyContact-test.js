import VerifyContact from '../../src/Auth/VerifyContact';
import React from 'react';
import AmplifyTheme from '../../src/AmplifyTheme';
import AuthPiece from '../../src/Auth/AuthPiece';
import { Header, Footer, InputRow, RadioRow, MessageRow, ButtonRow, Link } from '../../src/AmplifyUI';
import { Auth } from 'aws-amplify';

const acceptedStates = [
    'verifyContact'
];

const deniedStates = [
    'signIn',  
    'signedUp', 
    'signedOut',
    'forgotPassword',
    'signedIn',
    'confirmSignIn',
    'confirmSignUp',
    'signUp'
];

describe.only('VerifyContent test', () => {
    describe('render test', () => {
        test('render with accepted states', () => {
            const wrapper = shallow(<VerifyContact/>);
            for (var i = 0; i < acceptedStates.length; i += 1){
                wrapper.setProps({
                    authState: acceptedStates[i],
                    theme: 'theme',
                    authData: {
                    unverified: {
                        email: 'email@amazon.com',
                        phone_number: '+12345678901'
                        }
                    }   
                });
                expect(wrapper).toMatchSnapshot();
            }
        });

        test('render submitView if verifyAttr is set', () => {
            const wrapper = shallow(<VerifyContact/>);
            const props = {
                authState: 'verifyContact',
                theme: 'theme'
            }
            wrapper.setProps(props);
            wrapper.setState({ verifyAttr: true });

            expect(wrapper).toMatchSnapshot();
        });

        test('render verifyView with empty authData', () => {
            const wrapper = shallow(<VerifyContact/>);
            const props = {
                authState: 'verifyContact',
                theme: 'theme',
                authData: {}
            }
            wrapper.setProps(props);

            expect(wrapper).toMatchSnapshot();
        });

        test('render verifyView without phone_number and email', () => {
            const wrapper = shallow(<VerifyContact/>);
            const props = {
                authState: 'verifyContact',
                theme: 'theme',
                authData: {
                    unverified: {}
                }
            }
            wrapper.setProps(props);

            expect(wrapper).toMatchSnapshot();
        });

        test('render null with denied states', () => {
            const wrapper = shallow(<VerifyContact/>);
            
            for (var i = 0; i < deniedStates.length; i += 1){
                wrapper.setProps({
                    authState: deniedStates[i],
                    theme: 'theme'
                });

                expect(wrapper).toMatchSnapshot();
            }
        });

        test('hidden', () => {
            const wrapper = shallow(<VerifyContact/>);
            const props = {
                authState: 'verifyContact',
                theme: 'theme',
                hide: [VerifyContact]
            }
            wrapper.setProps(props);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('interaction test', () => {
        test('Link clicked', () => {
            const spyon = jest.spyOn(VerifyContact.prototype, 'changeState').mockImplementationOnce(() => {
                return;
            });
            const wrapper = shallow(<VerifyContact/>);
            const props = {
                authState: 'verifyContact',
                theme: 'theme'
            }
            wrapper.setProps(props);

            wrapper.find(Link).simulate('click');

           expect(spyon).toBeCalled();

            spyon.mockClear();
        });

        test('verifyView ButonRow clicked', () => {
            const spyon = jest.spyOn(VerifyContact.prototype, 'verify').mockImplementationOnce(() => {
                return;
            });
            const wrapper = shallow(<VerifyContact/>);
            const props = {
                authState: 'verifyContact',
                theme: 'theme',
                authData: {
                unverified: {
                    email: 'email@amazon.com',
                    phone_number: '+12345678901'
                    }
                }   
            }
            wrapper.setProps(props);
            
            wrapper.find(ButtonRow).at(0).simulate('click');

            //expect(spyon).toBeCalled();

            spyon.mockClear();
        });
    });

    describe('verify test', () => {
        test('happy case', async () => {
            const spyon = jest.spyOn(Auth, 'verifyCurrentUserAttribute').mockImplementationOnce(() => {
                return new Promise((res, rej) => {
                    res('data');
                });
            });
            const wrapper = shallow(<VerifyContact/>);
            const verifyContact = wrapper.instance();
            verifyContact.inputs = {
                contact: true,
                checkedValue: 'email'
            }

            await verifyContact.verify();
            expect(spyon).toBeCalledWith('email');

            spyon.mockClear();
        });

        test('no cantact', async () => {
            const spyon = jest.spyOn(VerifyContact.prototype, 'error').mockImplementationOnce(() => {
                return;
            });

            const wrapper = shallow(<VerifyContact/>);
            const verifyContact = wrapper.instance();
            verifyContact.inputs = {
                contact: null
            }

            await verifyContact.verify();
            expect(spyon).toBeCalledWith('Neither Email nor Phone Number selected');

            spyon.mockClear();
        });

        test('auth error', async () => {
            const spyon = jest.spyOn(Auth, 'verifyCurrentUserAttribute').mockImplementationOnce(() => {
                return new Promise((res, rej) => {
                    rej('err');
                });
            });
            const spyon2 = jest.spyOn(VerifyContact.prototype, 'error').mockImplementationOnce(() => {
                return;
            });

            const wrapper = shallow(<VerifyContact/>);
            const verifyContact = wrapper.instance();
            verifyContact.inputs = {
                contact: 'contact'
            }

            await verifyContact.verify();
            expect(wrapper.state('verifyAttr')).toBeNull();

            spyon.mockClear();
            spyon2.mockClear();
        });
    });

    describe('submit test', () => {
        test('happy case', async () => {
            const spyon = jest.spyOn(Auth, 'verifyCurrentUserAttributeSubmit').mockImplementationOnce(() => {
                return new Promise((res, rej) => {
                    res('data');
                });
            });
            const spyon2 = jest.spyOn(VerifyContact.prototype, 'changeState').mockImplementationOnce(() => {
                return;
            });

            const wrapper = shallow(<VerifyContact/>);
            const verifyContact = wrapper.instance();
            wrapper.setState({ verifyAttr: 'attr' });
            verifyContact.inputs = {
                code: 'code'
            }

            await verifyContact.submit();
            expect(spyon).toBeCalledWith('attr', 'code');

            spyon.mockClear();
            spyon2.mockClear();
        });

        test('auth error', async () => {
            const spyon = jest.spyOn(Auth, 'verifyCurrentUserAttributeSubmit').mockImplementationOnce(() => {
                return new Promise((res, rej) => {
                    rej('err');
                });
            });
            const spyon2 = jest.spyOn(VerifyContact.prototype, 'error').mockImplementationOnce(() => {
                return;
            });
            const spyon3 = jest.spyOn(VerifyContact.prototype, 'changeState').mockImplementationOnce(() => {
                return;
            });

            const wrapper = shallow(<VerifyContact/>);
            const verifyContact = wrapper.instance();
            verifyContact.inputs = {
                code: 'code'
            }

            await verifyContact.submit();
            expect(spyon3).not.toBeCalled();

            spyon.mockClear();
            spyon2.mockClear();
            spyon3.mockClear();
        });
    });
});