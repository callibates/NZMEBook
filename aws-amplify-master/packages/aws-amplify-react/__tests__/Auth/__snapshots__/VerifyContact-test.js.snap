// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifyContent test render test hidden 1`] = `""`;

exports[`VerifyContent test render test render null with denied states 1`] = `""`;

exports[`VerifyContent test render test render null with denied states 2`] = `""`;

exports[`VerifyContent test render test render null with denied states 3`] = `""`;

exports[`VerifyContent test render test render null with denied states 4`] = `""`;

exports[`VerifyContent test render test render null with denied states 5`] = `""`;

exports[`VerifyContent test render test render null with denied states 6`] = `""`;

exports[`VerifyContent test render test render null with denied states 7`] = `""`;

exports[`VerifyContent test render test render null with denied states 8`] = `""`;

exports[`VerifyContent test render test render submitView if verifyAttr is set 1`] = `
<FormSection
  theme="theme"
>
  <SectionHeader
    theme="theme"
  >
    Verify Contact
  </SectionHeader>
  <SectionBody
    theme="theme"
  >
    <MessageRow
      theme="theme"
    >
      Account recovery requires verified contact information
    </MessageRow>
    <div>
      <InputRow
        key="code"
        name="code"
        onChange={[Function]}
        placeholder="Code"
        theme="theme"
      />
      <ButtonRow
        onClick={[Function]}
        theme="theme"
      >
        Submit
      </ButtonRow>
    </div>
  </SectionBody>
  <SectionFooter
    theme="theme"
  >
    <Link
      onClick={[Function]}
      theme="theme"
    >
      Skip
    </Link>
  </SectionFooter>
</FormSection>
`;

exports[`VerifyContent test render test render verifyView with empty authData 1`] = `
<FormSection
  theme="theme"
>
  <SectionHeader
    theme="theme"
  >
    Verify Contact
  </SectionHeader>
  <SectionBody
    theme="theme"
  >
    <MessageRow
      theme="theme"
    >
      Account recovery requires verified contact information
    </MessageRow>
  </SectionBody>
  <SectionFooter
    theme="theme"
  >
    <Link
      onClick={[Function]}
      theme="theme"
    >
      Skip
    </Link>
  </SectionFooter>
</FormSection>
`;

exports[`VerifyContent test render test render verifyView without phone_number and email 1`] = `
<FormSection
  theme="theme"
>
  <SectionHeader
    theme="theme"
  >
    Verify Contact
  </SectionHeader>
  <SectionBody
    theme="theme"
  >
    <MessageRow
      theme="theme"
    >
      Account recovery requires verified contact information
    </MessageRow>
    <div>
      <ButtonRow
        onClick={[Function]}
        theme="theme"
      >
        Verify
      </ButtonRow>
    </div>
  </SectionBody>
  <SectionFooter
    theme="theme"
  >
    <Link
      onClick={[Function]}
      theme="theme"
    >
      Skip
    </Link>
  </SectionFooter>
</FormSection>
`;

exports[`VerifyContent test render test render with accepted states 1`] = `
<FormSection
  theme="theme"
>
  <SectionHeader
    theme="theme"
  >
    Verify Contact
  </SectionHeader>
  <SectionBody
    theme="theme"
  >
    <MessageRow
      theme="theme"
    >
      Account recovery requires verified contact information
    </MessageRow>
    <div>
      <RadioRow
        key="email"
        name="contact"
        onChange={[Function]}
        placeholder="Email"
        theme="theme"
        value="email"
      />
      <RadioRow
        key="phone_number"
        name="contact"
        onChange={[Function]}
        placeholder="Phone Number"
        theme="theme"
        value="phone_number"
      />
      <ButtonRow
        onClick={[Function]}
        theme="theme"
      >
        Verify
      </ButtonRow>
    </div>
  </SectionBody>
  <SectionFooter
    theme="theme"
  >
    <Link
      onClick={[Function]}
      theme="theme"
    >
      Skip
    </Link>
  </SectionFooter>
</FormSection>
`;
